### PR TITLE
fix: do not redact secret value when it is file path

### PR DESCRIPTION
**Pull request by zmstone**
Release version: v/e5.8.4

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [x] Changed lines covered in coverage report
- [x] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx/src/emqx_schema_secret.erl [hunk 1]
```diff
@@ -71,8 +71,8 @@ convert_secret(Secret, #{}) ->
     end.
 
 -spec wrap(source()) -> emqx_secret:t(t()).
-wrap(<<"file://", Filename/binary>>) ->
-    emqx_secret:wrap_load({file, Filename});
+wrap(<<"file://", _Filename/binary>> = Secret) ->
+    emqx_secret:wrap_load({file, Secret});
```
Comment (keynslug) [inline apps/emqx/src/emqx_schema_secret.erl:75]: Q: Why this change is needed?

File: apps/emqx/src/emqx_secret_loader.erl [hunk 2]
```diff
@@ -22,9 +22,13 @@
 
 -export_type([source/0]).
 
--type source() :: {file, file:filename_all()}.
+-type source() :: {file, string() | binary()}.
 
 -spec load(source()) -> binary() | no_return().
+load({file, <<"file://", Path/binary>>}) ->
+    file(Path);
+load({file, "file://" ++ Path}) ->
```
Comment (keynslug) [inline apps/emqx/src/emqx_secret_loader.erl:30]: Nit. RIght now it's not possible to wrap a `string()`, only `binary()`.

File: apps/emqx/src/emqx_schema_secret.erl (hunk 1 above)
Comment (zmstone) [reply apps/emqx/src/emqx_schema_secret.erl:75]: this is to keep the original prefix so there is no need to concatenate "file://" after calling `emqx_secret:term`.

**Review by keynslug — APPROVED** [review]

File: apps/emqx/src/emqx_schema_secret.erl (hunk 1 above)
Comment (keynslug) [reply apps/emqx/src/emqx_schema_secret.erl:75]: Just to clarify: the idea was to keep this `file://...` convention on the schema level only, so that lower-level code would not have to know about it.

File: apps/emqx/src/emqx_secret_loader.erl [hunk 3]
```diff
@@ -22,9 +22,13 @@
...
+    file(Path);
 load({file, Filename}) ->
     file(Filename).
```
[the remark this answers is not in the document]
Comment (zmstone) [reply apps/emqx/src/emqx_secret_loader.erl]: removed.

File: apps/emqx/src/emqx_secret_loader.erl (hunk 2 above)
Comment (zmstone) [reply apps/emqx/src/emqx_secret_loader.erl:30]: right. but I'd keep this clause here just in case.

**Review by keynslug — APPROVED** [review]

**Comment by emqxqa** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-5333)
